sql/date-and-time-types.md

```diff
@@ -0,0 +1,227 @@
+---
+title: date and time types
+category: user guide
+---
+
+# Date and time types
+
+TiDB supports the following data types to store temporal values: `DATE`, `TIME`, `DATETIME`, `TIMESTAMP`, and `YEAR`. Each of these types has its own range of valid values, and uses a zero value to indicate that it is an invalid value. In addition, the `TIMESTAMP` and `DATETIME` types can automatically generate new time values on modification.
+
+When dealing with date and time value types, please note: 
+
++ Although TiDB tries to interpret different formats, the date-portion must be in the format of year-month-day (for example, '1998-09-04'), rather than month-day-year or day-month-year.
++ If the year-portion of a date is specified as 2 digits, TiDB converts it based on the following rules:
+  - Values between 70 and 99 is converted to a value between 1970 and 1999
+  - Values between 00 and 69 is converted to a value between 2000 and 2069
++ If a numberic value is needed in the context, TiDB automatically converts the date or time value into a numeric type. For example,
+
+  ```sql
+  mysql> SELECT NOW(), NOW()+0, NOW(3)+0;
+  +---------------------+----------------+--------------------+
+  | NOW()               | NOW()+0        | NOW(3)+0           |
+  +---------------------+----------------+--------------------+
+  | 2012-08-15 09:28:00 | 20120815092800 | 20120815092800.889 |
+  +---------------------+----------------+--------------------+
+  ```
+  
++ TiDB may automatically convert invalid values or values beyond the supported range to a zero value of that type. This behavior is dependent on the SQL Mode set. For example,
+
+  ```sql
+  mysql> set @@sql_mode = '';                               
+  Query OK, 0 rows affected (0.00 sec)
+
+  mysql> show create table t1;
+  +-------+---------------------------------------------------------------------------------------------------------+
+  | Table | Create Table                                                                                            |
+  +-------+---------------------------------------------------------------------------------------------------------+
+  | t1    | CREATE TABLE `t1` (
+    `a` time DEFAULT NULL
+  ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin |
+  +-------+---------------------------------------------------------------------------------------------------------+
+  1 row in set (0.00 sec)
+  
+  mysql> insert into t1 values ('2090-11-32:22:33:44');
+  Query OK, 1 row affected, 1 warning (0.00 sec)
+  
+  mysql> select * from t1;
+  +----------+
+  | a        |
+  +----------+
+  | 00:00:00 |
+  +----------+
+  1 row in set (0.00 sec)
+  ```
+
++ Setting different SQL modes can change TiDB behaviors.
++ TiDB allows month or day in the columns of `DATE` and `DATETIME` to be zero value, for example, '2009-00-00' or '2009-01-00'. If this date type is to be calculated in a function, for example, in `DATE_SUB()` or `DATE_ADD()`, the result can be incorrect. 
```

Comment:
iirc, this one is also sql mode dependent (`NO_ZERO_IN_DATE`).  The description should suggest that it is not possible by default.